### PR TITLE
feat(xray-core): new xray features

### PR DESCRIPTION
**Pull request by ImMohammad20000**
- [x] add noise to hosts
- [x] add support for splithttp new variables

### CODE REVIEW

File: app/xray/config.py [hunk 1]
```diff
@@ -293,6 +293,8 @@ def _resolve_inbounds(self):
                     settings['scMaxConcurrentPosts'] = net_settings.get('scMaxConcurrentPosts',
                                                                         net_settings.get('maxConcurrentUploads', 100))
                     settings['scMinPostsIntervalMs'] = net_settings.get('scMinPostsIntervalMs', 30)
+                    settings['noSSEHeader'] = net_settings.get('noSSEHeader', False)
```
Comment (M03ED) [inline app/xray/config.py:296]: noSSEHeader:
(Server-only) Do not send the Content-Type: text/event-stream response header. Defaults to false (the header will be sent)
https://xtls.github.io/en/config/transports/splithttp.html#splithttpobject

File: app/db/models.py [hunk 2]
```diff
@@ -214,6 +214,7 @@ class ProxyHost(Base):
     is_disabled = Column(Boolean, nullable=True, default=False)
     mux_enable = Column(Boolean, nullable=False, default=False, server_default='0')
     fragment_setting = Column(String(100), nullable=True)
+    noise_setting = Column(String(100), nullable=True)
```
Comment (M03ED) [inline app/db/models.py:217]: noise its only for wireguard and currently its not supported

Comment (ImMohammad20000) [reply app/db/models.py:217]: No it work with splithttp h3 too

**Comment by ImMohammad20000** [comment]
> because noise str must be a bytestring, and bytestrings are too long, 100 characters for noise settings is not enough

is 300 character ok?

**Comment by ImMohammad20000** [comment]
> I captured another one and turned it to bytestring, it's 2k characters, i think it's better if you change the limit to 5000

Can you send a picture or type that here?

**Comment by ImMohammad20000** [comment]
I think 1000 characters is enough

**Comment by ImMohammad20000** [comment]
> Some UDP headers are up to 2k, how you think 1000 is enough?

I think it better to remove the character limit

**Comment by M03ED** [comment]
I think we should wait until they complete this feature then add it,

**Comment by ImMohammad20000** [comment]
I will update this to new syntax

**Comment by ImMohammad20000** [comment]
now you can use this syntax to add more than 1 noise
`rand:1-4,10-25&rand:1-4,10-25&base64:7nQBAAABAAAAAAAABnQtcmluZwZtc2VkZ2UDbmV0AAABAAE=,10-25`

**Comment by ImMohammad20000** [comment]
> There is a problem, if you put both maxConcurrency and maxConnections options, Xray will not get started
> Adding them all with value 0 is the problem, if maxConcurrency does not exist in the server config, it must not exist in client config too, creating maxConcurrency with value 0 is not equal to not existence of it
> and as already there is v2ray-settings template, there is no need to add support for xmux!

Is this problem stile exists in the latest version of xray core?

**Comment by ImMohammad20000** [comment]
Please merge this
@M03ED @SaintShit

**Comment by ImMohammad20000** [comment]
I'll add http transport in a separate pr

**Pull request merged**